test/unit/lib/plugins/aws/package/compile/events/iotFleetProvisioning/index.test.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+const chai = require('chai');
+const runServerless = require('../../../../../../../../utils/run-serverless');
+const templateBody = require('../../../../../../../../fixtures/iotFleetProvisioning/template.json');
+
+chai.use(require('chai-as-promised'));
+
+const { expect } = chai;
+
+describe('AwsCompileIoTFleetProvisioningEvents', () => {
+  const functionName = 'iotFleetProvisioningBasic';
+  let cfResources;
+  let naming;
+
+  before(async () => {
+    const { awsNaming, cfTemplate } = await runServerless({
+      fixture: 'function',
+      configExt: {
+        functions: {
+          iotFleetProvisioningBasic: {
+            handler: 'index.main',
+            events: [
+              {
+                iotFleetProvisioning: {
+                  templateBody,
+                  provisioningRoleArn: 'arn:aws:iam::123456789:role/provisioning-role',
+                },
+              },
+            ],
+          },
+        },
+      },
+      cliArgs: ['package'],
+    });
+    ({ Resources: cfResources } = cfTemplate);
+    naming = awsNaming;
+  });
+
+  it('should create corresponding template resource', () => {
+    const iotProvisioningTemplateResource =
+      cfResources[naming.getIotFleetProvisioningLogicalId(functionName)];
+    expect(iotProvisioningTemplateResource).to.deep.equal({
+      Type: 'AWS::IoT::ProvisioningTemplate',
+      Properties: {
+        Enabled: true,
+        PreProvisioningHook: {
+          TargetArn: {
+            'Fn::GetAtt': [naming.getLambdaLogicalId(functionName), 'Arn'],
+          },
+        },
+        ProvisioningRoleArn: 'arn:aws:iam::123456789:role/provisioning-role',
+        TemplateBody: JSON.stringify(templateBody),
+      },
+      DependsOn: [naming.getLambdaIotFleetProvisioningPermissionLogicalId(functionName)],
+    });
+  });
+  it('should create corresponding permission resource', () => {
+    const lambdaPermissionResource =
+      cfResources[naming.getLambdaIotFleetProvisioningPermissionLogicalId(functionName)];
+    expect(lambdaPermissionResource).to.deep.equal({
+      Type: 'AWS::Lambda::Permission',
+      Properties: {
+        FunctionName: {
+          'Fn::GetAtt': [naming.getLambdaLogicalId(functionName), 'Arn'],
+        },
+        Action: 'lambda:InvokeFunction',
+        Principal: 'iot.amazonaws.com',
+      },
+    });
+  });
+
+  describe('With enabled parameter', () => {
+    before(() =>
+      runServerless({
+        fixture: 'function',
+        configExt: {
+          functions: {
+            iotFleetProvisioningBasic: {
```

Comment:
Can we configure this test case in previous `runServerless` run? (as those runs are expensive ideally if there's minimal count of them)